### PR TITLE
make jira authentication Python3 compatible

### PR DESCRIPTION
**Pull request by tim-werner**
##### SUMMARY

Fixes problem with jira module and Python3

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/modules/web_infrastructure/jira.py

##### ANSIBLE VERSION

```
ansible 2.4.0.0
python version = 3.6.3 (default, Oct  6 2017, 11:37:43) [GCC 4.9.2]
```

##### ADDITIONAL INFORMATION

Full traceback before change
```
The full traceback is:
Traceback (most recent call last):
  File "/home/twerner/ansible_venv/lib/python3.6/base64.py", line 517, in _input_type_check
    m = memoryview(s)
TypeError: memoryview: a bytes-like object is required, not 'str'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/tmp/ansible_zjezlbg5/ansible_module_jira.py", line 431, in main
    ret = method(restbase, user, passwd, module.params)
  File "/tmp/ansible_zjezlbg5/ansible_module_jira.py", line 324, in fetch
    ret = get(url, user, passwd, params['timeout'])
  File "/tmp/ansible_zjezlbg5/ansible_module_jira.py", line 275, in get
    return request(url, user, passwd, timeout)
  File "/tmp/ansible_zjezlbg5/ansible_module_jira.py", line 247, in request
    auth = base64.encodestring('%s:%s' % (user, passwd)).replace('\n', '')
  File "/home/twerner/ansible_venv/lib/python3.6/base64.py", line 547, in encodestring
    return encodebytes(s)
  File "/home/twerner/ansible_venv/lib/python3.6/base64.py", line 534, in encodebytes
    _input_type_check(s)
  File "/home/twerner/ansible_venv/lib/python3.6/base64.py", line 520, in _input_type_check
    raise TypeError(msg) from err
TypeError: expected bytes-like object, not str
```

### CODE REVIEW

File: lib/ansible/modules/web_infrastructure/jira.py
Comment (sivel) [inline lib/ansible/modules/web_infrastructure/jira.py]: This line is incompatible with py2.6, and should either continue using printf style string formatting, or use element positions such as `{0}:{1}` instead of `{}:{}`

Additionally, for proper types, please utilize `ansible.module_utils._text.to_text`.

File: lib/ansible/modules/web_infrastructure/jira.py
Comment (sivel) [inline lib/ansible/modules/web_infrastructure/jira.py]: This block is unnecessary, when using `to_text` as described above.

**Review by sivel — DISMISSED** [review]

**Comment by tim-werner** [comment]
I did the requested changes and it would be very kind of you if you could have another look and approve the pull request when you think that all necessary points are addressed. Thx

File: lib/ansible/modules/web_infrastructure/jira.py
Comment (abadger) [inline lib/ansible/modules/web_infrastructure/jira.py]: Also import to_bytes (So we can use it below)

File: lib/ansible/modules/web_infrastructure/jira.py
Comment (abadger) [inline lib/ansible/modules/web_infrastructure/jira.py]: Use to_bytes instead of encode()
``` python
auth = to_text(base64.b64encode(to_bytes('{0}:{1}'.format(user, passwd), errors='surrogate_or_strict')))
```

Comment (abadger) [reply lib/ansible/modules/web_infrastructure/jira.py]: Why was replace('\n', '') removed?  If that was required before I don't see anything that would replace it now.

**Comment by abadger** [comment]
Made note of a couple problems before this can be merged.


needs_revision

File: lib/ansible/modules/web_infrastructure/jira.py
Comment (tim-werner) [reply lib/ansible/modules/web_infrastructure/jira.py]: `base64.encodestring(s) returns a string containing one or more lines of base64-encoded data always including an extra trailing newline ('\n')` which is not the case for b64encode and hence we do not need it anymore

**Comment by abadger** [comment]
Merged to devel and a backport PR opened for 2.5.1 https://github.com/ansible/ansible/pull/37800

**Pull request merged**
